lib/spack/spack/test/cmd/style.py

```diff
@@ -98,6 +98,25 @@ def test_changed_files(flake8_package):
     assert flake8_package in files
 
 
+def test_changed_files_from_git_rev_base(tmpdir, capfd):
+    """Test arbitrary git ref as base."""
+    git = which("git", required=True)
+    with tmpdir.as_cwd():
+        git("init", "-b", "main")
```

Comment:
`git init -b` is only supported since git 2.28. CentOS8 git is 2.27.

```suggestion
        git("init")
        git("checkout", "-b", "main")
```